### ISSUE TITLE
Extend na_ontap_security_config to specify server certificate to use

### ISSUE THREAD

**Issue by Jiawei0227** [ansible-collections/netapp.ontap#23]
### Summary

I am trying to translate the following command into an ansible module.
`ssl modify -vserver <vserver-name> -server-enabled true -client-enabled true -common-name <common-name> -serial <SN-from-trusted-CA-cert> -ca <cert-authority>`

And it seems the existing `na_ontap_security_config` cannot feed my need?

### Component Name

na_ontap_security_config

### Additional Information

```yaml (paste below)
- name: Modify SSL Security Config
  na_ontap_security_config:
    name: ssl
    vserver: xxx
    server_enabled: true
    client_enabled: true
    serial: xxx
    ca: xxx
    common_name: xxx
    <<: *ontap
```

**Comment by lonico** [comment]
I don't think this is something you need to do on recent versions of ONTAP.  Which version are you using?

Also, this would not belong to this module.   We would need a different one.
And there is currently no REST support for this
`security-ssl-modify
This ONTAPI call does not have an equivalent REST API call.`

**Comment by Jiawei0227** [comment]
I am using ONTAP 9.8. Basically I want to install my own CA to the SVM.

More context: I am setting up trident on my k8s cluster and try to follow the cert-based authentication in this doc: https://netapp-trident.readthedocs.io/en/stable-v21.01/kubernetes/operations/tasks/backends/ontap/ontap-nas/preparing.html#certificated-based-authentication

I am trying to convert all the commands into ansible module. But I couldnt find the one for ssl modify... So right now I have to use a na_ontap_command which isnt idea.

**Comment by lonico** [comment]
In my experience, this is not needed with 9.8.
I also have an example at
https://github.com/ansible-collections/netapp.ontap/blob/main/playbooks/examples/na_ontap_pb_install_SSL_certificate_REST.yml

**Comment by Jiawei0227** [comment]
Hey @lonico Thanks a lot for the example! It seems the example it only creates client_ca which does not add trusted CA certificate to the ONTAP cluster as describe in https://netapp-trident.readthedocs.io/en/stable-v21.01/kubernetes/operations/tasks/backends/ontap/ontap-nas/preparing.html#certificated-based-authentication as step 2? The type for that cert is server. And later I need to `ssl modify` to use that as server certificate to check client-ca that are signed by the same CA.

**Comment by lonico** [comment]
Yes, it's only an example.   Do you see any issue if you don't use `ssl modify`?    It should not be required.

**Comment by Jiawei0227** [comment]
I guess if not use ssl modify to choose the correct "Server Certificate Issuing CA", it will use the SVM default server certificate issuing CA? And in that case will it still work?

**Comment by lonico** [comment]
We reached out to the ONTAP team, and this could be handled through PATCH/svm/svms/{uuid}.  We'll look into it.

**Comment by Jiawei0227** [comment]
Thanks so much!

**Comment by lonico** [comment]
fixed in 21.18.0
